Pipelines 1.6.3 Release Notes (#45915)
openshift/openshift-docs: modules/op-release-notes-1-6.adoc

* Administrators can disable the creation of the `pipeline` service account for the entire cluster, and prevent privilege escalation by misusing the associated SCC, which is very similar to `anyuid`.

> * You can now configure feature flags and components by using the `TektonConfig` custom resource (CR), and the custom resources (CRs) for individual components, such as `TektonPipeline` and `TektonTriggers`. This level of granularity is useful for customization and testing of alpha features such as the Tekton OCI bundle for individual components.

Reviewer: Maybe "...helps you..."

** For implicit params:
*** Do not apply the `PipelineSpec` parameters to the `TaskRefs` object.
*** Disable implicit param behavior for the `Pipeline` objects.
// https://github.com/tektoncd/pipeline/releases/tag/v0.28.3

[id="release-notes-1-6-3_{context}"]
== Release notes for {pipelines-title} General Availability 1.6.3

[id="fixed-issues-1-6-3_{context}"]
=== Fixed issues

* Before this update, the {pipelines-title} Operator installed pod security policies from components such as Pipelines and Triggers. However, the pod security policies shipped as part of the components were deprecated in an earlier release. With this update, the Operator stops installing pod security policies from components. As a result, the following upgrade paths are affected:

> ** When you upgrade from {pipelines-shortname} 1.6.1 or 1.6.2 to {pipelines-shortname} 1.6.3, the existing pod security policies from components such as Pipelines and Triggers are deleted.

Reviewer: - Does upgrading delete the security policies from any component (first suggestion), or only the security policies from Pipelines and Triggers (second suggestion)? 
```suggestion
** Upgrading from {pipelines-shortname} 1.6.1 or 1.6.2 to {pipelines-shortname} 1.6.3 deletes the pod security policies, including those from the Pipelines and Triggers components.
```
```suggestion
** Upgrading from {pipelines-shortname} 1.6.1 or 1.6.2 to {pipelines-shortname} 1.6.3 deletes the pod security policies from the Pipelines and Triggers components.
```
Sometimes, if it doesn't add clarity, you can drop `existing` as redundant.

> ** When you upgrade from {pipelines-shortname} 1.5.x to 1.6.3, the existing pod security policies are retained. However, cluster administrators can delete them manually.

Reviewer: ```suggestion
** Upgrading from {pipelines-shortname} 1.5.x to 1.6.3 retains the pod security policies. As a cluster administrator, delete these pod security policies by using the `[TBD]` command.
```
- You can remove "However." It creates a false opposition with the first sentence.
- Do we need to clarify whether the policies are still in-effect or tell the administrator to delete them?

+
[NOTE]
====
When you upgrade to future releases, the {pipelines-title} Operator will automatically delete all obsolete pod security policies.
====
// https://issues.redhat.com/browse/SRVKP-2259

* Before this update, only cluster administrators could access pipeline metrics in the {product-title} console. With this update, users with other cluster roles also can access the pipeline metrics.
// https://issues.redhat.com/browse/SRVKP-2129

> * Before this update, multiple implementation issues in role-based access control (RBAC) caused rare problems during the installation and upgrade of components shipped with {pipelines-shortname} Operator. With this update, the reliability and consistency of installing various components of {pipelines-title} are improved.

Reviewer: ```suggestion
* Before this update, role-based access control (RBAC) issues with the {pipelines-shortname} Operator caused problems upgrading or installing components. This update improves the reliability and consistency of installing various {pipelines-title} components.
```

// https://issues.redhat.com/browse/SRVKP-2249

> * Before this update, setting the `clusterTasks` and `pipelineTemplates` fields to `false` in the `TektonConfig` CR resulted in the slow removal of cluster tasks and pipeline templates. With this update, the speed of lifecycle management of Tekton resources such as cluster tasks and pipeline templates has improved.

Reviewer: ```suggestion
* Before this update, setting the `clusterTasks` and `pipelineTemplates` fields to `false` in the `TektonConfig` CR slowed the removal of cluster tasks and pipeline templates. This update improves the speed of lifecycle management of Tekton resources such as cluster tasks and pipeline templates.
```

** For implicit params:
*** Do not apply the `PipelineSpec` parameters to the `TaskRefs` object.
*** Disable implicit param behavior for the `Pipeline` objects.
// https://github.com/tektoncd/pipeline/releases/tag/v0.28.3

[id="release-notes-1-6-3_{context}"]
== Release notes for {pipelines-title} General Availability 1.6.3

[id="deprecated-features-1-6-3_{context}"]
=== Deprecated features

* With this update, the {pipelines-title} Operator stops installing pod security policies from components, as the policies are deprecated. As a result, the following upgrade paths are affected:
** When you upgrade from {pipelines-shortname} 1.6.1 or 1.6.2 to {pipelines-shortname} 1.6.3, the existing pod security policies from components such as Pipelines and Triggers are deleted.

> ** When you upgrade from {pipelines-shortname} 1.5.x to 1.6.3, the existing pod security policies are retained. However, cluster administrators can delete them manually.

Reviewer 2: Ensure you follow this guideline mentioned here for Deprecated functionality: https://redhat-documentation.github.io/supplementary-style-guide/#release-notes

Author: This is not exactly a deprecated feature. This is actually a fixed issue that was overdue for fix after something was deprecated in an earlier release. Accordingly, I have restructured the sentence and included it under Fixed issues.

Reviewer 2: @sounix000 Thanks for the explanation. this helps :)

+
[NOTE]
====
When you upgrade to future releases, the {pipelines-title} Operator will automatically delete all obsolete pod security policies.
====
// https://issues.redhat.com/browse/SRVKP-2259

[id="fixed-issues-1-6-3_{context}"]
=== Fixed issues

> * Before this update, only cluster administrators could access pipeline metrics in the {product-title} console. With this update, users with other cluster roles can access the pipeline metrics.

Reviewer 2: ```suggestion
* Before this update, only cluster administrators could access pipeline metrics in the {product-title} console. With this update, users with other cluster roles also can access the pipeline metrics.
```

// https://issues.redhat.com/browse/SRVKP-2129

* With this update, the reliability and consistency of installing various components of {pipelines-title} are improved.
// https://issues.redhat.com/browse/SRVKP-2249

* With this update, the speed of lifecycle management of Tekton resources are improved.

> // https://issues.redhat.com/browse/SRVKP-2043

Reviewer 2: None of this mentions what was the case before the update. Please rewrite following the guideline prescribed here: https://redhat-documentation.github.io/supplementary-style-guide/#release-notes

> * Before this update, multiple implementation issues in RBAC caused rare problems during installation and upgrade of components shipped with {pipelines-shortname} Operator. With this update, the reliability and consistency of installing various components of {pipelines-title} are improved.

Reviewer 2: ```suggestion
* Before this update, multiple implementation issues in role-based access control (RBAC) caused rare problems during the installation and upgrade of components shipped with {pipelines-shortname} Operator. With this update, the reliability and consistency of installing various components of {pipelines-title} are improved.
```

> * Before this update, setting the `clusterTasks` and `pipelineTemplates` fields to `false` in the `TektonConfig` CR resulted in slow removal of cluster tasks and pipeline templates. With this update, the speed of lifecycle management of Tekton resources such as cluster tasks and pipeline templates have improved.

Reviewer 2: ```suggestion
* Before this update, setting the `clusterTasks` and `pipelineTemplates` fields to `false` in the `TektonConfig` CR resulted in the slow removal of cluster tasks and pipeline templates. With this update, the speed of lifecycle management of Tekton resources such as cluster tasks and pipeline templates has improved.
```
